### PR TITLE
ADBDEV-4799: Fix the environment for running test suite for gpdb7

### PR DESCRIPTION
**Pull request by KnightMurloc**
Fix the environment for running test suite for gpdb7

The container base has been changed from CentOS 7 to Rocky 8.8.
Dnf is now used instead of yum as a package manager.
Devtoolset-7 was removed since the necessary tools are available in the base
repository.
RSA1 key generation was removed since it's deprecated.
In docker-compose, the coordinator's name has been changed from mdw to cdw.
Containers also now use the init process to prevent zombie processes from
appearing.
Packages for llvm and clang are taken from the vault, since only clang-16 is
available in the main repository, and the build gpdb with llvm 16+ is not
supported.
Cherry-picked commit c24d7e60d7f8569522b2716b77e0905c91ad3d4f to fix gpfdist
tests with TLS 1.0 and TLS 1.1.
The attachment of gpdb logs to the allure report will be added as a separate patch.

### CODE REVIEW

**Review by RekGRpth — DISMISSED** [review]

File: arenadata/Dockerfile [hunk 1]
```diff
@@ -1,45 +1,55 @@
...
-    yum -y install apache-ivy libicu perl-ExtUtils-Embed perl-Env perl-JSON && \
-    yum -y install perl-IPC-Run perl-Test-Base libxslt-devel openldap-devel && \
-    yum clean all
+RUN dnf makecache && \
+    dnf -y install python39-devel && \
+    dnf -y install apr-devel bzip2 bzip2-devel expat-devel libcurl-devel && \
+    dnf -y install libevent-devel libuuid-devel libxml2-devel libyaml-devel libzstd-devel && \
+    dnf -y install openssl-devel pam-devel readline-devel snappy-devel libuv-devel && \
+    dnf -y install apache-ivy libicu perl-ExtUtils-Embed perl-Env perl-JSON && \
+    dnf -y install perl-IPC-Run perl-Test-Base libxslt-devel openldap-devel libpq-devel && \
```
Comment (andr-sokolov) [inline arenadata/Dockerfile]: Do we really need libpq-devel?

File: arenadata/Dockerfile [hunk 2]
```diff
@@ -1,45 +1,55 @@
...
-RUN yum makecache && yum update -y ca-certificates && \
-    rpm --import https://mirror.yandex.ru/centos/RPM-GPG-KEY-CentOS-7 && \
-    yum -y install epel-release java-1.8.0-openjdk-devel && \
-    yum -y install git iproute net-tools openssh-server rsync sudo time vim wget unzip \
-                   ant-junit autoconf bison cmake3 flex gperf indent jq libtool gcc-c++ \
-                   krb5-server krb5-workstation xerces-c-devel file && \
+RUN dnf -y install 'dnf-command(config-manager)' && \
+    dnf config-manager --set-enabled devel && dnf makecache && dnf update -y ca-certificates && \
+    rpm --import https://mirror.yandex.ru/rockylinux/RPM-GPG-KEY-Rocky-8 && \
+    dnf -y install epel-release java-1.8.0-openjdk-devel && \
```
Comment (andr-sokolov) [inline arenadata/Dockerfile]: Update java version
https://github.com/arenadata/gpdb/commit/c097366900c7699f6e10fe5f29a763792d34af41

File: arenadata/Dockerfile [hunk 3]
```diff
@@ -1,45 +1,55 @@
...
+RUN dnf makecache && \
+    dnf -y install python39-devel && \
+    dnf -y install apr-devel bzip2 bzip2-devel expat-devel libcurl-devel && \
+    dnf -y install libevent-devel libuuid-devel libxml2-devel libyaml-devel libzstd-devel && \
+    dnf -y install openssl-devel pam-devel readline-devel snappy-devel libuv-devel && \
+    dnf -y install apache-ivy libicu perl-ExtUtils-Embed perl-Env perl-JSON && \
+    dnf -y install perl-IPC-Run perl-Test-Base libxslt-devel openldap-devel libpq-devel && \
+    dnf clean all
+
+RUN rpm --import https://dl.rockylinux.org/pub/rocky/RPM-GPG-KEY-Rocky-8 && \
```
Comment (andr-sokolov) [inline arenadata/Dockerfile]: Are RPM-GPG-KEY-Rocky-8 here and above the same files?

File: arenadata/Dockerfile [hunk 4]
```diff
@@ -1,45 +1,55 @@
...
+    dnf install -y \
+                --repofrompath=devel_external,https://dl.rockylinux.org/vault/rocky/8.8/devel/x86_64/os \
+                llvm-libs-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                llvm-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                llvm-test-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                llvm-static-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                llvm-devel-15.0.7-1.module+el8.8.0+1144+0a4e73bd && \
+    dnf install -y \
+                --repofrompath=AppStream_external,https://dl.rockylinux.org/vault/rocky/8.8/AppStream/x86_64/os \
```
Comment (andr-sokolov) [inline arenadata/Dockerfile]: Why we have two repositories to install llvm and clang?

File: arenadata/Dockerfile [hunk 5]
```diff
@@ -1,45 +1,55 @@
...
 RUN ssh-keygen -t rsa -N "" -f /root/.ssh/id_rsa && \
     cat /root/.ssh/id_rsa.pub >> /root/.ssh/authorized_keys && \
     chmod 0600 /root/.ssh/authorized_keys && \
     echo -e "password\npassword" | passwd 2> /dev/null && \
-    { ssh-keyscan localhost; ssh-keyscan 0.0.0.0; } >> /root/.ssh/known_hosts && \
-    #
-    ssh-keygen -f /etc/ssh/ssh_host_key -N '' -t rsa1 && \
     ssh-keygen -f /etc/ssh/ssh_host_rsa_key -N '' -t rsa && \
     ssh-keygen -f /etc/ssh/ssh_host_dsa_key -N '' -t dsa && \
+    /sbin/sshd & sleep 5 && \
```
Comment (andr-sokolov) [inline arenadata/Dockerfile]: Is it possible to decrease changes here?

File: arenadata/Dockerfile [hunk 6]
```diff
@@ -1,45 +1,55 @@
...
+                llvm-libs-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                llvm-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                llvm-test-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                llvm-static-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                llvm-devel-15.0.7-1.module+el8.8.0+1144+0a4e73bd && \
+    dnf install -y \
+                --repofrompath=AppStream_external,https://dl.rockylinux.org/vault/rocky/8.8/AppStream/x86_64/os \
+                clang-resource-filesystem-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                clang-libs-15.0.7-1.module+el8.8.0+1144+0a4e73bd \
+                clang-15.0.7-1.module+el8.8.0+1144+0a4e73bd && \
```
Comment (andr-sokolov) [inline arenadata/Dockerfile:35]: How many compilers is it need to build gpdb?

File: arenadata/Dockerfile (hunk 3 above)
Comment (KnightMurloc) [reply arenadata/Dockerfile]: they are the same, so this line can be omitted.

File: arenadata/Dockerfile (hunk 6 above)
Comment (KnightMurloc) [reply arenadata/Dockerfile:35]: clang is required for an llvm-enabled build.
```
configure: error: clang not found, but required when compiling --with-llvm, specify with CLANG=
```

File: arenadata/Dockerfile (hunk 4 above)
Comment (KnightMurloc) [reply arenadata/Dockerfile]: yes, we can install all the necessary packages from one repository.

File: arenadata/Dockerfile (hunk 1 above)
Comment (KnightMurloc) [reply arenadata/Dockerfile]: packages from the repository are installed in the upstream.
https://github.com/greenplum-db/gpdb/blob/e7c2b1f14bb42a1018ac57d14f4436880e0a0515/README.Rhel-Rocky.bash#L5-L12

File: arenadata/Dockerfile [hunk 7]
```diff
@@ -1,45 +1,55 @@
...
-                   ant-junit autoconf bison cmake3 flex gperf indent jq libtool gcc-c++ \
-                   krb5-server krb5-workstation xerces-c-devel file && \
+RUN dnf -y install 'dnf-command(config-manager)' && \
+    dnf config-manager --set-enabled devel && dnf makecache && dnf update -y ca-certificates && \
+    rpm --import https://mirror.yandex.ru/rockylinux/RPM-GPG-KEY-Rocky-8 && \
+    dnf -y install epel-release java-1.8.0-openjdk-devel && \
+    dnf -y install git iproute net-tools openssh-server rsync sudo time vim wget unzip \
+                   ant-junit autoconf bison cmake3 flex gperftools indent jq libtool gcc-c++ \
+                   krb5-server krb5-workstation xerces-c-devel file netcat passwd lsof xz && \
     yum clean all
```
Comment (Stolb27) [inline arenadata/Dockerfile]: It's the last and only presence of yum in this file. Is it correct?

**Review by Stolb27 — COMMENTED** [review]
Please don't merge this PR. I think we may test CI on it.

**Comment by RekGRpth** [comment]
cherry-picked commits lost authorship

File: src/bin/gpfdist/regress/output/gpfdist_old_ssl.source [hunk 8]
```diff
@@ -1,3 +1,37 @@
+--
+-- GPFDISTS test cases
+--
+-- start_ignore
+drop external table if exists gpfdist_ssl_start;
+NOTICE:  table "gpfdist_ssl_start" does not exist, skipping
+drop external table if exists gpfdist_ssl_stop;
+NOTICE:  table "gpfdist_ssl_stop" does not exist, skipping
+-- end_ignore
+-- --------------------------------------
```
Comment (RekGRpth) [inline src/bin/gpfdist/regress/output/gpfdist_old_ssl.source:10]: ignore content in output is redundant

Comment (KnightMurloc) [reply src/bin/gpfdist/regress/output/gpfdist_old_ssl.source:10]: this is cherry pick. is it worth editing it?

Comment (RekGRpth) [reply src/bin/gpfdist/regress/output/gpfdist_old_ssl.source:10]: > this is cherry pick

OK

**Review by RekGRpth — DISMISSED** [review]

File: arenadata/Dockerfile [hunk 9]
```diff
@@ -1,45 +1,53 @@
...
+    dnf clean all
 
 # setup ssh configuration
 RUN ssh-keygen -t rsa -N "" -f /root/.ssh/id_rsa && \
     cat /root/.ssh/id_rsa.pub >> /root/.ssh/authorized_keys && \
     chmod 0600 /root/.ssh/authorized_keys && \
     echo -e "password\npassword" | passwd 2> /dev/null && \
-    { ssh-keyscan localhost; ssh-keyscan 0.0.0.0; } >> /root/.ssh/known_hosts && \
-    #
-    ssh-keygen -f /etc/ssh/ssh_host_key -N '' -t rsa1 && \
```
Comment (bandetto) [inline arenadata/Dockerfile:29]: Why was key generation removed?

File: arenadata/docker-compose.yaml [hunk 10]
```diff
@@ -4,19 +4,20 @@ version: "3"
...
     volumes:
       - "$PWD/ssh_keys/id_rsa:/home/gpadmin/.ssh/id_rsa"
       - "$PWD/ssh_keys/id_rsa.pub:/home/gpadmin/.ssh/id_rsa.pub"
       - "$PWD/allure-results:/tmp/allure-results"
     privileged: true
     sysctls:
       kernel.sem: 500 1024000 200 4096
+    init: true
     entrypoint: >
-      bash -c "set -e && tail -f /home/gpadmin/gpdb_src/README.md"
```
Comment (bandetto) [inline arenadata/docker-compose.yaml:19]: Is this line equivalent to `sleep infinity`? Does enabling init increase overhead?

File: arenadata/Dockerfile [hunk 11]
```diff
@@ -1,45 +1,53 @@
...
-RUN yum makecache && \
-    yum -y install python3-devel && \
-    yum -y install apr-devel bzip2-devel expat-devel libcurl-devel && \
-    yum -y install libevent-devel libuuid-devel libxml2-devel libyaml-devel libzstd-devel && \
-    yum -y install openssl-devel pam-devel readline-devel snappy-devel libuv-devel && \
-    yum -y install apache-ivy libicu perl-ExtUtils-Embed perl-Env perl-JSON && \
-    yum -y install perl-IPC-Run perl-Test-Base libxslt-devel openldap-devel && \
-    yum clean all
+RUN dnf makecache && \
+    dnf -y install python39-devel && \
```
Comment (bandetto) [inline arenadata/Dockerfile:14]: I tried to run tests in the container before this patch and got a error about python version. Why a different version was needed and how this issue was fixed?

File: arenadata/docker-compose.yaml (hunk 10 above)
Comment (KnightMurloc) [reply arenadata/docker-compose.yaml:19]: yes, it is equivalent, but in rocky 8 it does not work the way it does in centos 7. the task of the command `bash -c "set -e && tail -f /home/gpadmin/gpdb_src/README.md"` is to start an infinite wait and that the process with pid 1 is bash (for the correct completion of "orphaned" processes) but in rocky 8, pid 1 gets tail (possibly due to some new bash optimizations) and because of that, zombie processes begin to appear.
I don't think the init process will add an overhead.

File: arenadata/Dockerfile (hunk 11 above)
Comment (KnightMurloc) [reply arenadata/Dockerfile:14]: gpdb 7 migrated from python 2 to python 3

File: arenadata/Dockerfile (hunk 9 above)
Comment (KnightMurloc) [reply arenadata/Dockerfile:29]: rsa1 is deprecated and an attempt to generate it fails.

File: arenadata/Dockerfile (hunk 11 above)
Comment (bandetto) [reply arenadata/Dockerfile:14]: I meant that the package `python3-devel` contained the wrong version of Python (`python3.6`) and now the package `python39-devel` has the correct version of Python.

**Review by bandetto — DISMISSED** [review]

File: arenadata/Dockerfile (hunk 6 above)
Comment (KnightMurloc) [reply arenadata/Dockerfile:35]: clang pulls gcc as a dependency.

**Comment by BenderArenadata** [comment]
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1047327

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1047328

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1047779

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1048016

**Comment by BenderArenadata** [comment]
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1048013

**Comment by BenderArenadata** [comment]
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1048015

**Comment by BenderArenadata** [comment]
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1049307

**Comment by BenderArenadata** [comment]
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1051594

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1052385

**Comment by BenderArenadata** [comment]
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1052384

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1053666

**Comment by BenderArenadata** [comment]
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1053665

**Comment by BenderArenadata** [comment]
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1053663

**Comment by BenderArenadata** [comment]
Failed job Regression tests with Postgres on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1053662

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1054872

**Comment by BenderArenadata** [comment]
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1054871

**Comment by BenderArenadata** [comment]
Failed job Regression tests with Postgres on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1054868

**Comment by BenderArenadata** [comment]
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1054869

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1057738

**Comment by BenderArenadata** [comment]
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1057737

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests 7: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1062178

**Comment by BenderArenadata** [comment]
Failed job Behave tests 7: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1062177

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests 7: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1062821

**Comment by BenderArenadata** [comment]
Failed job Behave tests 7: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1062820

**Comment by RekGRpth** [comment]
Complete the description of the PR in accordance with the requirements (title, width 80, ...)

**Comment by RekGRpth** [comment]
Why second commit does not correspond to https://github.com/arenadata/gpdb/commit/c097366900c7699f6e10fe5f29a763792d34af41?

**Comment by KnightMurloc** [comment]
> Why second commit does not correspond to https://github.com/arenadata/gpdb/commit/c097366900c7699f6e10fe5f29a763792d34af41?

What do you mean? The list of changes in them is the same.

**Comment by RekGRpth** [comment]
> > Why second commit does not correspond to [c097366](https://github.com/arenadata/gpdb/commit/c097366900c7699f6e10fe5f29a763792d34af41)?
> 
> What do you mean? The list of changes in them is the same.

```diff
2c2
< index bb6fa5335f65..5c7339d9d150 100644
---
> index eab478296971..666ed5ac8b74 100644
5c5
< @@ -3,9 +3,9 @@ FROM centos:centos7 as base
---
> @@ -12,9 +12,9 @@ RUN sed -i 's/\(override_install_langs*\)/# \1/' /etc/yum.conf && \
14c14
<                     krb5-server krb5-workstation xerces-c-devel file && \
---
>                     krb5-server krb5-workstation xerces-c-devel && \
17,18c17,18
< @@ -15,7 +15,7 @@ RUN yum makecache && \
<      yum -y install apr-devel bzip2-devel expat-devel libcurl-devel && \
---
> @@ -25,7 +25,7 @@ RUN yum makecache && \
>      yum -y install apr-devel apr-util-devel bzip2-devel expat-devel libcurl-devel && \
23,25c23,25
<      yum -y install perl-IPC-Run perl-Test-Base libxslt-devel openldap-devel && \
<      yum clean all
<  
---
>      yum -y install perl-IPC-Run perl-Test-Base libxslt-devel && \
>      yum -y install libzstd-static && \
>      # The last python 2.7 pip
```

**Comment by KnightMurloc** [comment]
in this regard, yes, they are different because the dockerfile for 6X and 7X are different. But the list of changes (removed packages) is the same.

**Review by RekGRpth — APPROVED** [review]

File: arenadata/Dockerfile [hunk 12]
```diff
@@ -1,45 +1,49 @@
...
 
 # Install some basic utilities and build tools
-RUN yum makecache && yum update -y ca-certificates && \
-    rpm --import https://mirror.yandex.ru/centos/RPM-GPG-KEY-CentOS-7 && \
-    yum -y install epel-release java-1.8.0-openjdk-devel && \
-    yum -y install git iproute net-tools openssh-server rsync sudo time vim wget unzip \
-                   ant-junit autoconf bison cmake3 flex gperf indent jq libtool gcc-c++ \
-                   krb5-server krb5-workstation xerces-c-devel file && \
-    yum clean all
+RUN dnf -y install 'dnf-command(config-manager)' && dnf config-manager --set-enabled devel && dnf makecache && dnf update -y ca-certificates && \
```
Comment (Stolb27) [inline arenadata/Dockerfile:4]: Is this changes required for vault repo? Maybe combine it together to simplify revert at further?

File: arenadata/readme.md [hunk 13]
```diff
@@ -22,16 +22,13 @@ We need to execute [../concourse/scripts/ic_gpdb.bash](../concourse/scripts/ic_g
...
-  -e MAKE_TEST_COMMAND=-k PGOPTIONS='-c optimizer=on' installcheck-world \
-  --sysctl kernel.sem=500 1024000 200 4096 gpdb7_regress:latest \
-  scl enable devtoolset-7 llvm-toolset-7 \
+  -e MAKE_TEST_COMMAND="-k PGOPTIONS='-c optimizer=on' installcheck-world" \
+  --sysctl "kernel.sem=500 1024000 200 4096" gpdb7_regress:latest \
   /home/gpadmin/gpdb_src/concourse/scripts/ic_gpdb.bash
 ```
 
 * we need to modify `MAKE_TEST_COMMAND` environment variable to run different suite. e.g. we may run test againt Postgres optimizer or ORCA with altering `PGOPTIONS` environment variable;
-* we need to run container as `--privileged` to run debugger inside it
```
Comment (Stolb27) [inline arenadata/readme.md:32]: Why do you consider this statement as redundant? Are you able to run gdb inside an unprivileged container?

File: arenadata/readme.md [hunk 14]
```diff
@@ -89,15 +86,10 @@ bash arenadata/scripts/run_behave_tests.bash gpstart gpstop
 ```
 
 
-Tests use `allure-behave` package and store allure output files in `allure-results` folder
-**NOTE** that `allure-behave` has too old a version because it is compatible with `python2`.
-Also, the allure report for each failed test has gpdb logs attached files. See `gpMgmt/test/behave_utils/arenadata/formatter.py`
```
Comment (Stolb27) [inline arenadata/readme.md:94]: Why did you remove attachment generation?

**Review by Stolb27 — COMMENTED** [review]
Please provide valid commit messages and squash fixup commits as far as this PR will be merged with rebase without squashing.

File: arenadata/Dockerfile (hunk 12 above)
Comment (KnightMurloc) [reply arenadata/Dockerfile:4]: this is necessary to add a devel repository from which some packages are taken. this is not related to llvm.

File: arenadata/readme.md (hunk 13 above)
Comment (KnightMurloc) [reply arenadata/readme.md:32]: yes

File: arenadata/readme.md (hunk 14 above)
Comment (KnightMurloc) [reply arenadata/readme.md:94]: I did not remove the generation of attachment. I just updated the description and left only the first line. it is marked as deleted since I added a dot at the end.

File: arenadata/readme.md (hunk 13 above)
Comment (Stolb27) [reply arenadata/readme.md:32]: Really?

```
[root@ac4d63bf750d gpadmin]# gdb -p 13079 --batch --ex 'bt 10'
ptrace: Operation not permitted.
No stack.
```

File: arenadata/readme.md (hunk 14 above)
Comment (Stolb27) [reply arenadata/readme.md:94]: As was mentioned in the PR description:

> The attachment of gpdb logs to the allure report will be added as a separate patch.

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests 7: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1069566

**Comment by BenderArenadata** [comment]
Failed job Behave tests 7: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1069565

**Review by Stolb27 — APPROVED** [review]
